04.ls/ls.rb

```diff
@@ -1,30 +1,53 @@
 #!/usr/bin/env ruby
 # frozen_string_literal: true
 
-NUMBER_OF_COL = 3
+require 'optparse'
+
+NUMBER_OF_COL_MAX = 3
 SPACE_WIDTH = 2
 
-def get_filenames(target_path)
-  Dir.glob('*', base: target_path)
+def search_option(argv)
+  opt = OptionParser.new
+  option_params = {}
+  opt.on('-a') { |v| option_params[:a] = v }
+  arguments_without_options = opt.parse(argv)
```

Comment:
この変数名、長い上にあまり抽象化できてないですね。。
オプションを取り除いたあとに残る情報はファイルパスになるはずなので、

```suggestion
  paths = opt.parse(argv)
```

とか

```suggestion
  file_paths = opt.parse(argv)
```

みたいな名前の方がいいと思います。

Reply:
なるほど、確かに言われてみるとわかりづらい名前ですね。
変数名の付け方がまだまだだと感じます。
提案頂いた`file_paths`に変更させて頂きます。